bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/autoupdate/internal/AutoUpdateProviderDelegate.java

```diff
@@ -20,26 +37,185 @@
+                    Metadata metadata = getMetadata(item.getName());
+                    metadataSet.add(metadata);
+                }
+            }
+        }
+        return metadataSet;
+    }
+
+    @Override
+    public void removeProviderChangeListener(ProviderChangeListener<Metadata> listener) {
+        this.listeners.remove(listener);
+    }
+
+    private void refreshItemUpdateVetos() {
+        Set<String> newVetos = new HashSet<>();
+        synchronized (itemUpdateVetos) {
+            itemUpdateVetos.clear();
+            for (Item item : itemRegistry.getAll()) {
+                for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+                    Boolean autoUpdate = provider.autoUpdate(item.getName());
+                    if (Boolean.FALSE.equals(autoUpdate)) {
+                        newVetos.add(item.getName());
+                    }
+                }
+            }
+
+            // find the removed ones
+            Set<String> removedVetos = new HashSet<>(itemUpdateVetos);
+            removedVetos.removeAll(newVetos);
+            for (String itemName : removedVetos) {
+                if (itemUpdateVetos.contains(itemName)) {
+                    Metadata md = getMetadata(itemName);
+                    for (ProviderChangeListener<Metadata> listener : listeners) {
+                        listener.removed(this, md);
+                    }
+                }
+            }
+
+            // find the added ones
+            Set<String> addedVetos = new HashSet<>(newVetos);
+            addedVetos.removeAll(itemUpdateVetos);
+            for (String itemName : addedVetos) {
+                notifyAboutAddedMetadata(itemName);
             }
+            itemUpdateVetos = newVetos;
         }
-        return null;
+    }
+
+    private void notifyAboutAddedMetadata(String itemName) {
+        if (itemUpdateVetos.contains(itemName)) {
+            Metadata md = getMetadata(itemName);
+            for (ProviderChangeListener<Metadata> listener : listeners) {
+                listener.added(this, md);
+            }
+        }
+    }
+
+    private void notifyAboutRemovedMetadata(String itemName) {
+        for (ProviderChangeListener<Metadata> listener : listeners) {
+            listener.removed(this, getMetadata(itemName));
+        }
+    }
+
+    private Metadata getMetadata(String itemName) {
+        return new Metadata(new MetadataKey(AUTOUPDATE_KEY, itemName), "false", null);
+    }
+
+    @Override
+    public void added(Item element) {
+        String itemName = element.getName();
+        refreshVetoForItem(itemName);
+    }
+
+    private void refreshVetoForItem(String itemName) {
+        synchronized (itemUpdateVetos) {
+            boolean removed = itemUpdateVetos.remove(itemName);
+            for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+                Object autoUpdate = provider.autoUpdate(itemName);
```

Comment:
In other places we at least use a `Boolean`...

Reply:
true